great_expectations/core/expectation_suite.py

```diff
@@ -669,29 +672,204 @@ def add_expectation(
+        ] = self._get_expectations_by_domain_using_accessor_method(
+            domain_type=MetricDomainTypes.TABLE.value,
+            accessor_method=self.get_table_expectations,
+        )
+        expectation_configurations_by_domain.update(
+            self._get_expectations_by_domain_using_accessor_method(
+                domain_type=MetricDomainTypes.COLUMN.value,
+                accessor_method=self.get_column_expectations,
+            )
+        )
+        expectation_configurations_by_domain.update(
+            self._get_expectations_by_domain_using_accessor_method(
+                domain_type=MetricDomainTypes.COLUMN_PAIR.value,
+                accessor_method=self.get_column_pair_expectations,
+            )
+        )
+        expectation_configurations_by_domain.update(
+            self._get_expectations_by_domain_using_accessor_method(
+                domain_type=MetricDomainTypes.MULTICOLUMN.value,
+                accessor_method=self.get_multicolumn_expectations,
+            )
+        )
+        return expectation_configurations_by_domain
+
+    def get_grouped_and_ordered_expectations_by_expectation_type(
+        self,
+    ) -> List[ExpectationConfiguration]:
+        """
+        Returns "ExpectationConfiguration" list, grouped by "expectation_type", in predetermined designated order.
+        """
+        table_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_table_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        column_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_column_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        column_pair_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_column_pair_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        multicolumn_expectation_configurations: List[ExpectationConfiguration] = sorted(
+            self.get_multicolumn_expectations(),
+            key=lambda element: element["expectation_type"],
+        )
+        return (
+            table_expectation_configurations
+            + column_expectation_configurations
+            + column_pair_expectation_configurations
+            + multicolumn_expectation_configurations
+        )
+
     def get_table_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of table expectations."""
-        return [
-            e
-            for e in self.expectations
-            if e.expectation_type.startswith("expect_table_")
-        ]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type() == MetricDomainTypes.TABLE,
+                self.expectations,
+            )
+        )
+
+        expectation_configuration: ExpectationConfiguration
```

Comment:
You don't explicitly need this since `expectation_configurations` is typed.

Reply:
This is our convention.  I am not sure what is the best way to evolve it, but happy to figure it out together -- uniformly.

Reply:
Agreed that we should have and stick to a convention. I'm fine with leaving this in. My preference is to annotate when needed but if one can observe the type close by and mypy can infer then let mypy infer. Of course that's not very precise and we don't currently run mypy (though it's coming to parts of our codebase soon!).

Reply:
This may all be wrong, but my justification was that at some point we might want to make interfaces using a typed language (Java, Scala), so this was to get us into the habit for that.  For now, this is just a style/practice based effort.  It would be nice to come up with "real" requirements/motivation at some point, and stick to them.  Thanks!